modules/consensus/processed_block.go

```diff
@@ -115,11 +103,20 @@ func (bn *blockNode) targetAdjustmentBase() *big.Rat {
 	// The target is converted to a big.Rat to provide infinite precision
 	// during the calculation. The big.Rat is just the int representation of a
 	// target.
-	timePassed := bn.block.Timestamp - windowStart.block.Timestamp
+	timePassed := pb.Block.Timestamp - windowStart.Block.Timestamp
 	expectedTimePassed := types.BlockFrequency * windowSize
 	return big.NewRat(int64(timePassed), int64(expectedTimePassed))
 }
 
+// setChildTarget computes the target of a blockNode's child. All children of a node
+// have the same target.
+func (pb *processedBlock) setChildTarget(db *setDB) {
```

Comment:
I agree, and passing the database here does seem unintuitive here. Unfortunately targetAdjustmentBase currently uses the database to find the timestamp of the block 512 earlier and this seemed like the simplest way to give it that.